tests/ftpserver.pl

```diff
@@ -828,7 +828,7 @@ sub MAIL_smtp {
         # Validate the from address (only <> and a valid email address inside
         # <> are allowed, such as <user@example.com>)
         if ((!$from) || (($from ne "<>") && ($from !~
-            /^<([a-zA-Z0-9._%+-]+)\@([a-zA-Z0-9.-]+).([a-zA-Z]{2,4})>$/))) {
+            /^<([a-zA-Z0-9._%+-]+)\@(([a-zA-Z0-9-]+)\.)+([a-zA-Z]{2,4})>$/))) {
```

Comment:
Ok, the escaping looks correct but was the removal from the []-group intentional?

Reply:
> Ok, the escaping looks correct but was the removal from the []-group intentional?
> 
> Update: ah, yes because you add an extra parentheses layer and a + around it... It seems you could have simplified that to:

Yeah that was intentional as you saw ;-) I appreciate I didn't need to include the + after the new group but it does allow for sub-domains.

> `/^<([a-zA-Z0-9._%+-]+)\@([a-zA-Z0-9.-]+)([a-zA-Z]{2,4})>$/`

Unfortunately both this and the original allow the following which I don't believe is valid:

`<bad@example......com>`

Reply:
Thinking about this further: does it matter? The server could just as well accept whatever we send it, as we just use it to verify that curl sends what we want it to...

Reply:
I actually see no point in having the server verify that field at all - we could simply by removing the check completely. If we want to verify that curl can deal with a 501 response to a `MAIL FROM` command, we already have other means for doing that!